src/markdown.c

```diff
@@ -769,8 +769,33 @@ char_entity(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t max_
 		if (!is_valid_numeric_entity(entity_val))
 			return 0;
 	} else {
-		if (!is_allowed_named_entity((const char *)data, end))
+		/* Lookup the entity in the named entity table. */
+		const struct html_entity *entity = is_allowed_named_entity((const char *)data, end);
+		if (!entity)
 			return 0;
+		if (entity->codepoints[0]) {
```

Comment:
Huh, are there cases where the entity is in the lookup table but it doesn't have any codepoints?